app/src/main/java/org/fossasia/openevent/general/discount/DiscountDao.kt

```diff
@@ -0,0 +1,20 @@
+package org.fossasia.openevent.general.discount
+
+import android.arch.persistence.room.Dao
+import android.arch.persistence.room.Insert
+import android.arch.persistence.room.OnConflictStrategy.REPLACE
+import android.arch.persistence.room.Query
+import io.reactivex.Single
+import org.fossasia.openevent.general.auth.DiscountCode
+
+@Dao
+interface DiscountDao {
+    @Insert(onConflict = REPLACE)
+    fun insertDiscountCode(discountCode: DiscountCode)
```

Comment:
DiscountCode should have Relation and foreign key with Event

Reply:
Still not done

Reply:
we already have a relations ship with EventId class , i suppose foreign key is not needed now

Reply:
It's absolutely needed

Reply:
look at the end of the file there is Relation with the event

Reply:
Relation can't ever be in DAO. It has to be in model. Take a look at SocialLink model. I think you are confusing JSONAPI Relationship with a conditional select query